### PR TITLE
feat: tables

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] | Fix RM-9021 |
| :--------------------: | :---------: |

## 🧰 Changes

MDX Tables!

I also took the opportunity to start an MDX components doc, and more excitingly, a nicer compilation error!

## ✅ TODO

After poking at this a bunch, I think we should punt on doing feature detection. It seems best that that be done as a singular after we get everything else working.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

File: index.tsx [hunk 1]
```diff
@@ -85,10 +88,7 @@ export const reactTOC = (text: string, opts = {}) => {
 };
 
 export const mdx = (tree: any, opts = {}) => {
-  return remark()
-    .use(remarkMdx)
-    .data({ toMarkdownExtensions: [{ extensions: [compilers] }] })
-    .stringify(tree, opts);
+  return remark().use(remarkMdx).use(remarkGfm).use(compilers).stringify(tree, opts);
```
Comment (kellyjosephprice) [inline index.tsx:91]: @jenn, just found out the way I was configuring custom compilers wouldn't override existing ones.

File: errors/mdx-syntax-error.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { VFileMessage } from 'vfile-message';
+
+export default class MdxSyntaxError extends SyntaxError {
+  original: VFileMessage = null;
+
+  constructor(error: VFileMessage, doc: string) {
+    const { message, line, column, url } = error;
+
+    const messages = [
+      `Uh oh! We ran into a syntax error at { line: ${line}, column: ${column} }, please see this url for more details: ${url}`,
```
Comment (jennspencer) [inline errors/mdx-syntax-error.ts:10]: image

File: docs/tables.md [hunk 3]
```diff
@@ -35,25 +36,32 @@ Tables have been simplified to mirror a more standard implementation. We've also
...
  */
-.markdown-body .rdmd-table th {}
-.markdown-body .rdmd-table td {}
+.markdown-body .rdmd-table th {
+}
+.markdown-body .rdmd-table td {
+}
 ```
 
 <style>
```
Comment (jennspencer) [inline docs/tables.md:67]: doesn't look like `acorn` is a big fan of this `<style>` tag:

image

but haaaay nice error messaging!

Comment (kellyjosephprice) [reply docs/tables.md:67]: It's not so much the script, but css. I imagine that MDX looked at trying to detect and parse css, and just table flipped

**Review by jennspencer — APPROVED** [review]
it's so beautiful 🥺

**Pull request merged**